### DISCUSSION TITLE
Defining a Variable as Output

### DISCUSSION THREAD

**Discussion by Faedon** [Consensys/gnark#601] [Q&A] [answered]
Hello, 

I need to use a variable inside the Define function as an input to another circuit. I basically need to define a variable as "signal output X" like in circom language. I have researched through the docs and the examples in the repo, but couldn't find an alias to output. It seems like only public and secret inputs are definable.

Is it possible to define a variable as output of the circuit? If not what else can i use?

Thank you for the answers!

**Comment by gbotrel** [comment, marked as answer]
Hi -- with gnark, you only have 1 circuit; to call "subcomponents", you just call functions.

**Reply by Faedon** [reply]
For example:

```
func (circuit *CubicCircuit) Define(api frontend.API) error {
	x3 := api.Mul(circuit.X, circuit.X, circuit.X)
	api.AssertIsEqual(circuit.Y, api.Add(x3, circuit.X, 5))
	return nil
}
```

I want to use x3 variable outside my circuit, not the return of a function.

**Reply by ivokub** [reply]
A circuit allows you to assert some relation between the inputs. It doesn't output a value.

If you want to use the input to the circuit somewhere else, then you can use the same value you are providing as a witness to the circuit.

```go
type CubicCircuit struct {
    X frontend.Variable
    Y frontend.Variable
}

func (circuit *CubicCircuit) Define(api frontend.API) error {
	x3 := api.Mul(circuit.X, circuit.X, circuit.X)
	api.AssertIsEqual(circuit.Y, api.Add(x3, circuit.X, 5))
	return nil
}

func main() {
    witness := Circuit{X: 3, Y: 35}
    // code for creating proof and verifying
    fmt.Println(witness.X) // use value X outside of circuit
}
```

**Reply by Faedon** [reply]
Oh, thank you for the answer!
